src/lib/operators/jit_operator/operators/jit_read_tuples.cpp

```diff
@@ -40,6 +42,18 @@ void JitReadTuples::before_query(const Table& in_table, JitRuntimeContext& conte
       });
     }
   }
+
+  if (_row_count_expression) {
```

Comment:
Do I understand this block correctly in that is has nothing to do with reading the tuples but you needed a place in which to evaluate the limit expression?

Reply:
Yes, the alternative approach would be to call a `before_query` function on every operator before execution. Then this code could be placed within the `JitLimit` operator. Should I change it to this approach that.

Reply:
No, just wanted to make sure I understood it correctly. A comment should be sufficient.